src/System.Text.Json/tests/WritableJsonApiTests.cs

```diff
@@ -625,5 +660,187 @@ public static void TestAquiringAllProperties()
+            {
+                // regular send:
+                Mailbox.SendAllEmployeesData(employeesToSend);
+
+                // modified elements send:
+                JsonNode modifiableDocument = JsonElement.DeepCopy(employeesToSend);
+                var employees = modifiableDocument as JsonObject;
+                employees.Add(EmployeesDatabase.GetNextEmployee());
+
+                JsonDocument modifiedEmployeesToSend = JsonDocument.Parse(employees);
+                Mailbox.SendAllEmployeesData(modifiedEmployeesToSend);
+            }
+        }
+
+        /// <summary>
+        /// Parsing right to JsonNode if user knows data will be modified
+        /// </summary>
+        [Fact]
+        public static void TestParsingToJsonNode()
+        {
+            string jsonString = @"
+            {
+                ""employee1"" : 
+                {
+                    ""name"" : ""Ann"",
+                    ""surname"" : ""Predictable"",
+                    ""age"" : 30,                
+                },
+                ""employee2"" : 
+                {
+                    ""name"" : ""Zoe"",
+                    ""surname"" : ""Coder"",
+                    ""age"" : 24,                
+                }
+            }";
+
+            JsonObject employees = JsonNode.Parse(jsonString) as JsonObject;
+            employees.Add(EmployeesDatabase.GetNextEmployee());
+            Mailbox.SendAllEmployeesData(JsonDocument.Parse(employees));
+        }
+
+        /// <summary>
+        /// Copying JsoneNode
+        /// </summary>
+        [Fact]
+        public static void TestCopyingJsonNode()
+        {
+            JsonObject employee = EmployeesDatabase.GetManager();
+            JsonNode employeeCopy = JsonElement.DeepCopy(employee);
+        }
+
+        /// <summary>
+        /// Checking IsImmutable property
+        /// </summary>
+        [Fact]
+        public static void TestIsImmutable()
+        {
+            JsonElement employee = Mailbox.RetrieveEmployeeData();
+            if(!employee.IsImmutable)
+            {
+                employee = JsonDocument.Parse(employee).RootElement;
+            }
+
+            // Perform operations requiring high performance
+            EmployeesDatabase.PerformHeavyOperations(employee);
+        }
+
+        /// <summary>
+        /// Sorting string JsonArray
+        /// </summary>
+        [Fact]
+        public static void TestSortingJsonArray()
+        {
+            var employees = new JsonArray
+            {
+                "Smith John",
+                "Predictable Ann",
+                "Coder Zoe",
+                "Shadow Byron"
+            };
+
+            employees.Sort();
+            
+            // Create medical appointments for employees in alphabetical order
+            foreach (JsonNode employee in employees)
+            {
+                if(employee is JsonString person)
+                    HealthCare.CreateMedicalAppointment(person.Value);
```

Comment:
Style: Unbraced block statements are (effectively) only allowed in the root method scope.  (The rule says it a different way, which is "braces are required in statements inside other braced statements"... and the foreach has braces)